modular_skyrat/modules/modular_items/lewd_items/code/lewd_arousal/climax.dm

```diff
@@ -0,0 +1,161 @@
+
+	if(!client?.prefs?.read_preference(/datum/preference/toggle/erp/autocum) && manual != TRUE)
+		return
+
+	if(has_status_effect(/datum/status_effect/climax_cooldown) || !client?.prefs?.read_preference(/datum/preference/toggle/erp/sex_toy))
+		return
+
+	if(HAS_TRAIT(src, TRAIT_NEVERBONER) || has_status_effect(/datum/status_effect/climax_cooldown) || (!has_vagina() && !has_penis()))
+		visible_message(span_purple("[src] twitches, trying to cum, but with no result."), \
+			span_purple("You can't have an orgasm!"))
+		return TRUE
+
+	// Reduce pop-ups and make it slightly more frictionless (lewd).
+	var/climax_choice = has_penis()? CLIMAX_PENIS : CLIMAX_VAGINA
+
+	if(manual)
+		var/list/genitals = list()
+		if(has_vagina())
+			genitals.Add(CLIMAX_VAGINA)
+			if(has_penis())
+				genitals.Add(CLIMAX_PENIS)
+				genitals.Add(CLIMAX_BOTH)
+		else if(has_penis())
+			genitals.Add(CLIMAX_PENIS)
+		climax_choice = tgui_alert(src, "You are climaxing, choose which genitalia to climax with.", "Genitalia Preference!", genitals)
+
+	switch(gender)
+		if(MALE)
+			playsound(get_turf(src), pick('modular_skyrat/modules/modular_items/lewd_items/sounds/final_m1.ogg',
+										'modular_skyrat/modules/modular_items/lewd_items/sounds/final_m2.ogg',
+										'modular_skyrat/modules/modular_items/lewd_items/sounds/final_m3.ogg'), 50, TRUE, ignore_walls = FALSE)
+		if(FEMALE)
+			playsound(get_turf(src), pick('modular_skyrat/modules/modular_items/lewd_items/sounds/final_f1.ogg',
+										'modular_skyrat/modules/modular_items/lewd_items/sounds/final_f2.ogg',
+										'modular_skyrat/modules/modular_items/lewd_items/sounds/final_f3.ogg'), 50, TRUE, ignore_walls = FALSE)
+
+	var/self_orgasm = FALSE
+	var/self_their = p_their()
+
+	if(climax_choice == CLIMAX_PENIS || climax_choice == CLIMAX_BOTH)
+		var/obj/item/organ/external/genital/penis/penis = getorganslot(ORGAN_SLOT_PENIS)
+		if(!getorganslot(ORGAN_SLOT_TESTICLES)) //If we have no god damn balls, we can't cum anywhere... GET BALLS!
+			visible_message(span_userlove("[src] orgasms, but nothing comes out of [self_their] penis!"), \
+				span_userlove("You orgasm, it feels great, but nothing comes out of your penis!"))
+
+		else if(is_wearing_condom())
+			var/obj/item/clothing/sextoy/condom/condom = get_item_by_slot(LEWD_SLOT_PENIS)
+			condom.condom_use()
+			visible_message(span_userlove("[src] shoots [self_their] load into the [condom], filling it up!"), \
+				span_userlove("You shoot your thick load into the [condom] and it catches it all!"))
+
+		else if(!is_bottomless() && penis.visibility_preference != GENITAL_ALWAYS_SHOW)
+			visible_message(span_userlove("[src] cums inside [self_their] clothes!"), \
+				span_userlove("You shoot your load, but you weren't naked, so you mess up your clothes!"))
+			self_orgasm = TRUE
+
+		else
+			var/list/interactable_inrange_humans = list()
+
+			for(var/mob/living/carbon/human/iterating_human in view(1, src))
+				if(iterating_human == src)
+					continue
```

Comment:
```suggestion
			for(var/mob/living/carbon/human/iterating_human in oview(1, src))
```

Reply:
oview() excludes the center atom

Reply:
Hahaha, old code leftover, can't believe I missed that one!